extension/src/time_vector.rs

```diff
@@ -115,6 +119,55 @@ pub fn unnest<'a>(
     )
 }
 
+/// Util function to convert from *const ::std::os::raw::c_char to String
+/// TimestampTz -> *const c_char -> &CStr -> &str -> String
+pub fn timestamptz_to_string(time: pg_sys::TimestampTz) -> Result<String, Utf8Error> {
+    let char_ptr = unsafe { timestamptz_to_str(time) };
+    let c_str = unsafe { CStr::from_ptr(char_ptr) };
+    c_str.to_str().map(|s| s.to_owned())
+}
+
+#[pg_extern(immutable, schema = "toolkit_experimental", parallel_safe)]
+pub fn format_timevector<'a>(series: Timevector_TSTZ_F64<'a>, format_string: String) -> String {
+    let mut context = Context::new();
+    let mut times: Vec<String> = Vec::new();
+    let mut values: Vec<String> = Vec::new();
+    if series.has_nulls() {
+        for (i, point) in series.iter().enumerate() {
+            times.push(timestamptz_to_string(point.ts).unwrap());
+            if series.is_null_val(i) {
+                values.push("null".to_string())
+            } else {
+                match point.val.to_string().as_ref() {
+                    "NaN" | "inf" | "-inf" | "Infinity" | "-Infinity" => {
+                        panic!("All values in the series must be finite")
+                    }
+                    x => values.push(x.to_string()),
+                }
+            }
+        }
+    } else {
+        // optimized path if series does not have any nulls, but might have some NaNs/infinities
+        for point in series {
+            times.push(timestamptz_to_string(point.ts).unwrap());
+            match point.val.to_string().as_ref() {
+                "NaN" | "inf" | "-inf" | "Infinity" | "-Infinity" => {
+                    panic!("All values in the series must be finite")
+                }
+                x => values.push(x.to_string()),
+            }
+        }
+    }
+    context.insert("TIMES", &times);
+    context.insert("VALUES", &values);
+    match format_string.as_ref() {
+	"plotly" =>     Tera::one_off("{\"times\": {{ TIMES | json_encode(pretty=true) | safe  }}, \"vals\": {{ VALUES | json_encode(pretty=true) | safe }}}", &context, true).expect("Failed to create plotly template"),
```

Comment:
Not sure how I feel about overloading the postgres facing function be either a user supplied format or some known magic string.  I'd prefer this to be an undocumented usage, and for us to proved a user facing `to_plotly` which is essentially just `format_output('plotly')`

Reply:
Swapping so that to_plotly() is its own separate function